src/engine-actions/validate-node-config.js

```diff
@@ -1,13 +1,78 @@
-const { getInfo } = require('../lnd-actions')
+const {
+  getInfo,
+  genSeed
+} = require('../lnd-actions')
 
+/**
+ * CODE 12 for gprc is equal to 'unimplemented'
+ * @see https://github.com/grpc/grpc-go/blob/master/codes/codes.go
+ * @constant
+ * @type {Number}
+ * @default
+ */
+const UNIMPLEMENTED_SERVICE_CODE = 12
+
+/**
+ * Rough estimate if the lnd instance's wallet is unlocked or not
+ *
+ * @private
+ * @param {grpc.Client} Lightning grpc client
+ * @return {Boolean} isEngineUnlocked
+ */
+async function isEngineUnlocked (client) {
+  try {
+    // If the call to `getInfo` succeeds, then we can assume that our LND instance
+    // is unlocked and functional
+    await getInfo({ client })
+    return true
+  } catch (e) {
+    // CODE 12 for grpc is equal to 'unknown service' or an error type of
+    // unimplemented.
+    // In GRPC Unimplemented indicates operation is not implemented or not
+    // supported/enabled in this specific service.
+    if (e.code && e.code === UNIMPLEMENTED_SERVICE_CODE) {
+      return false
+    }
+
+    // We return true, because the error is unrelated to a service not being implemented
+    // and the user will now have to troubleshoot further
+    return true
+  }
+}
 /**
  * Validates this engine's configuration against the node it is
  * hooked up to.
  *
  * @function
- * @return {Boolean} Whether the configuration matches the node
+ * @return {True} Node is configured correctly
+ * @throws {Error} Failed to validate an unlocked engine
+ * @throws {Error} LND has no chains configured
+ * @throws {Error} LND can only support one active chain at a time
+ * @throws {Error} Mismatched configuration if chain is different than engine configuration
  */
 async function validateNodeConfig () {
+  const isUnlocked = await isEngineUnlocked(this.client)
+
+  // We need to set the `isUnlocked` variable here so that end users can call the engine
+  // and validate if the engine is available or not
+  this.isUnlocked = isUnlocked
```

Comment:
This is what we will need to do, because validateNodeConfig can only run on a node that has a wallet (more info https://github.com/sparkswap/lnd-engine/pull/106#discussion_r231765013)